### PR TITLE
chore: cleanup appveyor definitions for not running jobs which is already run with GHA & add docker info/version commands

### PR DESCRIPTION
**Pull request by mndeveci**
#### Which issue(s) does this change fix?
N/A

#### Why is this change necessary?
Some of the checks that we do in GHA is also run with Appveyor jobs and they have custom configuration rather than using the ones from `Makefile`. 

#### How does it address the issue?
- Removing the tests/checks from Appveyor that we already run with GHA
- Cleaning up some Appveyor setup where it was repeated couple of times

#### What side effects does this change have?
N/A

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: appveyor-windows.yml [hunk 1]
```diff
@@ -135,6 +136,10 @@ install:
 
   - "IF DEFINED BY_CANARY ECHO Logging in Public ECR && aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws"
 
+build_script:
+  - "docker system prune -a -f"
```
Comment (super132) [inline appveyor-windows.yml:140]: Is it a Windows specific step? Why do we need to prune on Windows?

File: appveyor-ubuntu.yml [hunk 2]
```diff
@@ -206,10 +200,6 @@ for:
         - configuration: LocalZipTerraformBuildInContainerIntegTesting
 
     test_script:
-      # Pre-dev Tests
-      - "pip install -e \".[pre-dev]\""
```
Comment (jfuss) [inline appveyor-ubuntu.yml:210]: I think we need this or we need to move `ruff` into the `requirements/dev.txt`. I don't know why we have a "pre-dev" file.

**Review by jfuss — CHANGES_REQUESTED** [review]

**Comment by sriram-mv** [comment]
Assuming that linting is covered through `Makefile` targets being invoked with this removal.

File: appveyor-ubuntu.yml (hunk 2 above)
Comment (mndeveci) [reply appveyor-ubuntu.yml:210]: As far as I understand that `pre-dev.txt` is imported at the top of the `dev.txt` https://github.com/aws/aws-sam-cli/blob/develop/requirements/dev.txt#L1

File: appveyor-windows.yml (hunk 1 above)
Comment (super132) [reply appveyor-windows.yml:140]: Do we need to do the same for Ubuntu then?

File: appveyor-ubuntu.yml (hunk 2 above)
Comment (jfuss) [reply appveyor-ubuntu.yml:210]: Well would you look at that... lol

**Review by jfuss — APPROVED** [review]

**Review by sriram-mv — APPROVED** [review]

**Pull request merged**